System/Console/ANSI/Windows/Emulator/Codes.hs

```diff
@@ -0,0 +1,83 @@
+{-# OPTIONS_HADDOCK hide #-}
+
+module System.Console.ANSI.Windows.Emulator.Codes
+    (
+      -- * Cursor movement by character
+      cursorUpCode, cursorDownCode, cursorForwardCode, cursorBackwardCode
+
+      -- * Cursor movement by line
+    , cursorUpLineCode, cursorDownLineCode
+
+      -- * Directly changing cursor position
+    , setCursorColumnCode, setCursorPositionCode
+
+      -- * Clearing parts of the screen
+    , clearFromCursorToScreenEndCode, clearFromCursorToScreenBeginningCode
+    , clearScreenCode, clearFromCursorToLineEndCode
+    , clearFromCursorToLineBeginningCode, clearLineCode
+
+      -- * Scrolling the screen
+    , scrollPageUpCode, scrollPageDownCode
+
+      -- * Select Graphic Rendition mode: colors and other whizzy stuff
+    , setSGRCode
+
+      -- * Cursor visibilty changes
+    , hideCursorCode, showCursorCode
+
+      -- * Changing the title
+    , setTitleCode
+    ) where
+
+import System.Console.ANSI.Types
+
+cursorUpCode, cursorDownCode, cursorForwardCode, cursorBackwardCode :: Int -- ^ Number of lines or characters to move
+                                                                    -> String
+cursorUpCode _       = ""
+cursorDownCode _     = ""
+cursorForwardCode _  = ""
+cursorBackwardCode _ = ""
+
+cursorDownLineCode, cursorUpLineCode :: Int -- ^ Number of lines to move
+                                     -> String
+cursorDownLineCode _ = ""
```

Comment:
Are you saying that the `*Code` variables could be considered part of the "behaviour"? I guess this is true, and on the second reading, I see that it is explicitly mentioned in the docs:

>  On Windows these codes will always be the empty string, so it is possible to use them portably for e.g. coloring console output on the understanding that you will only see colors if you are running on a Unix-like operating system.

On the other hand:

1. Exposing two versions of the codes, empty and non-empty, is rather ugly.
2. I am not sure if many people use this "feature".

Let's leave it the way you did it for now, until we have a better idea.